htdocs/core/class/html.form.class.php

```diff
@@ -1664,8 +1664,27 @@ public function selectcontacts($socid, $selected = '', $htmlname = 'contactid',
 		}
 		$sql .= " WHERE sp.entity IN (".getEntity('socpeople').")";
 		if ($socid > 0 || $socid == -1) {
-			$sql .= " AND sp.fk_soc = ".((int) $socid);
-		}
+			$sql .= " AND (sp.fk_soc = ".((int) $socid);
+		}
+		// Show contact of parent
+    	if (!empty($conf->global->CONTACT_SHOW_ALSO_PARENT_COMPANY_CONTACTS)) {
+    		require_once DOL_DOCUMENT_ROOT.'/societe/class/societe.class.php';
+			$soc = new Societe($db);
+			$result = $soc->fetch($socid);
+			if ($result > 0 && !empty($soc->parent)) {
+				$sql .= ' OR sp.fk_soc = '.$soc->parent;
+			}
+    	}
+		// Show contacts of thirdparties with same parent
+    	if (!empty($conf->global->CONTACT_SHOW_ALSO_COMPANY_CONTACTS_WITH_SAME_PARENT)) {
+    		require_once DOL_DOCUMENT_ROOT.'/societe/class/societe.class.php';
+			$soc = new Societe($db);
+			$result = $soc->fetch($socid);
+			if ($result > 0 && !empty($soc->parent)) {
+				$sql .= ' OR sp.fk_soc IN (SELECT rowid FROM '.MAIN_DB_PREFIX.'societe WHERE parent = '.$soc->parent.')';
+			}
+    	}
+		$sql .= ')';
```

Comment:
A parenthesis is closed in any case, but it is open only in one case. This is surely a bug.
Also can you check your IDE to set tabs to be tabs instead of spaces ? This generated trouble on merging and code analysis.

Reply:
I was testing an IDE